main/acle.md

```diff
@@ -4836,6 +4838,47 @@ if such a restoration is necessary. For example:
    }
 ```
 
+## `__arm_agnostic`
+
+A function with the `__arm_agnostic` [keyword attribute](#keyword-attributes) must
+preserve the architectural state that is specified by its arguments when such
+state exists at runtime. The function is otherwise unconcerned with the contents
+of this state.
+
+The `__arm_agnostic` [keyword attribute](#keyword-attributes) applies to
+**function types** and accepts the following arguments:
+
+```"sme_za_state"```
+
+*   If the function is defined and PSTATE.ZA is available, the definition must
```

Comment:
Since the AAPCS64 patch (rightly) creates a three-way choice between private, shared & agnostic, we should update the description under “Mapping to the Procedure Call Standard” section.  I think the text here should then say something like:

> This attribute affects the ABI of object code functions, as described under …  It is the compiler's responsibility to ensure that such object code functions honor the ABI requirements.

and then delete the following two bullet points.  AFAICT, the attribute has no effect on the behaviour of the abstract machine outside of inline asms.